src/components/Navbar/Navbar.tsx

```diff
@@ -24,10 +25,33 @@ export const Navbar: React.FC<NavbarProps> = ({
   const topicMapSectionLabel = useL10n("navbarTopicMapSectionLabel");
   const notesSectionLabel = useL10n("navbarNotesSectionLabel");
   const helpSectionLabel = useL10n("navbarHelpSectionLabel");
+  const progressBarLabel = useL10n("progressBarLabel");
+  const progressPercentageLabel = useL10n("progressPercentageLabel");
   const deleteAllNotesText = useL10n("deleteNotesConfirmationWindowLabel");
   const deleteAllNotesConfirmText = useL10n("deleteNotesConfirmLabel");
   const deleteAllNotesDenyText = useL10n("deleteNotesDenyLabel");
   const userData = getUserData();
+  const [progressBarValue, setProgressBarValue] = React.useState<number>(0);
+  const totalNotesToComplete = topicMapItems.filter(
+    item => item.dialog && item.dialog.hasNote,
```

Comment:
Nit: Use optional chaining wherever feasible.

```suggestion
    item => item.dialog?.hasNote,
```

Reply:
Should we wrap this in a `React.memo` to only render whenever `topicMapItems` changes?

Reply:
⚠️ We also need to check arrow items' notes

Reply:
Yes! We can do that, although topicMapItems should not change for the end user dynamically afaik. But I agree that using the hook might be a safer and more efficient option here :)